service/history/decisionHandler.go

```diff
@@ -237,6 +238,13 @@ func (handler *decisionHandlerImpl) handleDecisionTaskFailed(
 		RunId:      common.StringPtr(token.RunID),
 	}
 
+	if request.GetCause().Equals(workflow.DecisionTaskFailedCauseWorkflowWorkerUnhandledFailure) {
+		handler.logger.Info("Timer double fire issue",
```

Comment:
make the log info timer_double_fire_issue - just so that it's easily searchable. Also, please emit a metric as well so that there can be a alarm on this